### ISSUE TITLE
Loss function doesn't decrease

### ISSUE THREAD

**Issue by eliwilner** [AntonMu/TrainYourOwnYOLO#70]
Hi @AntonMu 

I've tried you model last night in AWS EC2 machine.
My dataset contains 6 class, 1000 images pee class. I did 100 epochs and this morning I saw that the loss function is stuck around a value of ~35 after 50 epochs.
Any suggestions how to improve that?

Thanks again

**Comment by AntonMu** [comment]
Hi @eliwilner - is that one the first stage or the second stage? 

There are two parts of the training, in the first stage only the last layers are trained with the rest frozen and in the second more layers are unfrozen. Usually, you would see a larger loss decrease once you reach the second stage. You can also use tensorboard to track the progress and get a better idea of what is going on. 

Also, it looks like you have a relatively large number of training images and you could generally unfreeze more layers even for the first stage. 

Finally, it could be that your images are very different from ImageNet1000 (this is what the network is pre-trained on) and thus it cannot "transfer" that knowledge to your case. Again, unfreezing more layers should help.

**Comment by silvestre139** [comment]
@eliwilner  May I ask you what configuration you used to run the model in EC2? I tried ubuntu Deep learning Base and Ubuntu Deep Learning Conda, and I am facing different errors in both environments. (Suggestion will be really appreciated, thank you in advance!)

**PS**
The loss function behaves as AntonMu explained after your epoch surpass 51, the loss function will start showing a significant decrease, also you can take a look at your data augmentation, I did not take a look at the YOLO process in this specific section, but depending on the Data Augmentation process implemented for your images this could be a pro, or a BIG CON.

**Comment by AntonMu** [comment]
Also as a side note - the current implementation does automatic
augmentation. So even if you don't do it yourself it will use some.

**Comment by eliwilner** [comment]
@silvestre139 
I'm using Deep Learning AMI with Ubuntu 18.04.

Did you try it. Also, I'm running on p2.8xlarge
Let me know if it helped

**Comment by eliwilner** [comment]
> @eliwilner May I ask you what configuration you used to run the model in EC2? I tried ubuntu Deep learning Base and Ubuntu Deep Learning Conda, and I am facing different errors in both environments. (Suggestion will be really appreciated, thank you in advance!)
> 
> **PS**
> The loss function behaves as AntonMu explained after your epoch surpass 51, the loss function will start showing a significant decrease, also you can take a look at your data augmentation, I did not take a look at the YOLO process in this specific section, but depending on the Data Augmentation process implemented for your images this could be a pro, or a BIG CON.

Hi. So I'm currently on epoch 54 and still don't see any significant changes. Even the number of steps for train increased at stage two. Is this still a normal behavior? Were you able to get good results?
